### PR TITLE
feat(file-uploader): add FileUploader to @carbon/styles

### PR DESCRIPTION
**Pull request by tay1orjones**
Refs #8899

Brings `FileUploader` to the `@carbon/styles` package

#### Changelog

**New**

- include refactored file added to `/styles`, including a test

### CODE REVIEW

File: packages/styles/scss/components/file-uploader/_file-uploader.scss [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+@use '../../utilities/focus-outline' as *;
+@use '../../utilities/high-contrast-mode' as *;
+@use '../../utilities/visually-hidden' as *;
+
+@use '../button';
+@use '../form';
+@use '../loading';
+
+// TODO - how should we resolve this? Should we keep it? Is the '@use type' above enough?
+// @use '../../globals/scss/vendor/@carbon/elements/scss/type/styles';
```
Comment (tay1orjones) [inline packages/styles/scss/components/file-uploader/_file-uploader.scss]: Is the `@use '../../type' as *;` above enough to remove this?

Comment (tay1orjones) [reply packages/styles/scss/components/file-uploader/_file-uploader.scss]: Sounds like the answer to this is yes, the vendor stuff can be removed. I've pushed an update to fix.

**Review by joshblack — CHANGES_REQUESTED** [review]
Could you include the story files too @tay1orjones? Just to double-check that the styles work 👀  (sorry we forgot to include this in the template 🤦 )

**Comment by tay1orjones** [comment]
@joshblack Yes! Thank you for the reminder. I took a stab at not only bringing them over but refactoring them to use the new storybook 6 args, argTypes, and controls.

**Comment by joshblack** [comment]
@tay1orjones seems like a weird storybook issue coming up 🤔

**Review by tw15egan — APPROVED** [review]
The controls look ✨ wonderful ✨  💯

**Comment by tay1orjones** [comment]
@joshblack I'm not sure what happened, but it's up and working now since I updated with latest from `main`.

**Review by joshblack — APPROVED** [review]

**Pull request merged**